File: 3 - Searching & Sorting/Sorting/bubbleSort.cpp
#include <iostream>
using namespace std;

/* Bubble sort
--> It works by repeately swapping the adjacent elements
--> In every round the largest value takes their correct place

T.C : O(n^2)
S.C : O(1)
*/

int sort(int arr[],int n){

    for(int i=0;i<n-1;i++){
        for(int j=0;j<n-1;j++){
            if(arr[j]>arr[j+1]){ // for sort in decreasing order arr[j]<arr[j+1]
                swap(arr[j],arr[j+1]);
            }
        }
    }

    for(int i=0;i<n;i++){
        cout << arr[i] << " ";
    }
}

int main(){

    int arr[]={7,5,8,3,4,2};
    int n=6;

    sort(arr,n);
}

File: 2 - Arrays/1D Arr/bubble_sort.java
public class bubble_sort{

    public void bubble(int arr[], int n) {
		for(int i=0;i<n-1;i++){
            //for round 1 to n-1;
            for(int j=0;j<n-1;j++){
                if(arr[j]>arr[j+1]){
                    int temp=arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=temp;
                }
            }
        }
        for(int i=0;i<n;i++){
            System.out.print(arr[i]+" ");
        }
	}
    public static void main(String[] args) {
        int []arr={0,2,0,1,2,2,1,0,0};
        bubble_sort s=new bubble_sort();
        s.bubble(arr,arr.length);
    }
}